리팩토링 진행중

WhiteHyun opened the pull request:
## Changes 📝

- Timer 관련 코드는 `Delegate Pattern`을  이용하여 모듈화 작업함
- 뷰 컨트롤러의 이름을 역할에 맞게 수정
- 폴더 구조화 작업 진행

```
 📁 Egg Timer
 ┣ 📝 README
 ┃ 
 ┣ 📂 Constants
 ┃ ┣ 📝 Color
 ┃ ┣ 📝 UserDefaults
 ┃ ┣ 📝 NotificationName
 ┃ ┗ 📝 UserNotification
 ┃ 
 ┣ 📂 Resource
 ┃ ┗ Assets 
 ┃ 
 ┗ 📂 Source
   ┣ 📝 AppDelegate
   ┣ 📝 SceneDelegate
   ┃
   ┣ 📂 Model
   ┃ ┗ 📝 EggTimeManager
   ┃
   ┣ 📂 View
   ┃ ┗ 📝 ClockLayer
   ┃
   ┗ 📂 ViewController
     ┣ 📝 BaseViewController
     ┃ 
     ┣ 📂 Timer
     ┃ ┗ 📝 TimerViewController
     ┃
     ┗ 📂 Settings
       ┣ 📝 SettingsViewController
       ┗ 📝 SettingTableViewCell
```



## Screenshots 📸

WhiteHyun committed:
:pencil2: VC 이름 변경

ViewController -> TimerViewController

WhiteHyun committed:
:truck: 파일 정리

WhiteHyun committed:
:fire: 공백 일부 제거

eung7 commented on Egg Timer/Constants/UserDefaults.swift:
```diff
@@ -6,11 +6,10 @@ extension UserDefaults {
     /// 시계를 변경할 때 사용되는 키 값입니다.
-    static let switchClock         = "switchClockKey"
-    
+    static let switchClock = "switchClockKey"
```
안녕하세요 WhiteHyun

isGeekCode commented:
do you know Kim-chi?

isGeekCode commented on Egg Timer/Constants/UserDefaults.swift:
```diff
@@ -6,11 +6,10 @@ extension UserDefaults {
     /// 앱이 백그라운드로 들어가고 나서 시간을 저장할 때 사용되는 키 값입니다.
-    static let didEnterBackgroundDate     = "didEnterBackgroundDate"
+    static let didEnterBackgroundDate = "didEnterBackgroundDate"
     
```
키값을 enum으로 하셨네요. 저같은 경우는 클래스로 생성하는데 특별한 이유가 있을까요

isGeekCode replied:
이런 식으로 사용하는 거군요

isGeekCode reviewed (commented):
고생하셨어요

isGeekCode commented on Egg Timer/Constants/UserDefaults.swift:
```diff
@@ -6,11 +6,10 @@ extension UserDefaults {
     /// 시계를 변경할 때 사용되는 키 값입니다.
-    static let switchClock         = "switchClockKey"
-    
+    static let switchClock = "switchClockKey"
```
흠. 괜찮군요

WhiteHyun replied:
감사합니다!

isGeekCode commented on Egg Timer/Constants/UserDefaults.swift:
```diff
@@ -6,11 +6,10 @@ extension UserDefaults {
   enum Keys {
     
```
여기에 코드 작성해주세요!

WhiteHyun replied:
왜요?

isGeekCode commented on Egg Timer/Source/SceneDelegate.swift:
```diff
@@ -29,7 +29,7 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
     // view controller instance 가져옴
-    let vc = ViewController()
+    let vc = TimerViewController()
     
```
여기는 이런식으로 하시면 안돼요! ㅎㅎ

isGeekCode reviewed (changes requested):
새로 보냅니다

WhiteHyun committed:
:truck: ClockLayer 소스코드를 View 폴더로 옮김

WhiteHyun committed:
:recycle: 타이머 모델 구현하여 타이머 의존성 낮춤

- EggTimeManager 클래스를 구현하고, delegate 패턴을 이용하여 상호의존을 줄임.

WhiteHyun committed:
:truck: 소스코드 각 역할에 맞춰 옮김

- 버튼별, label별, 등으로 나누어 코드를 구조화
- lazy var 버튼의 경우 viewDidLoad에 있는 addTarget 명령어를 then 클로저 안으로 옮김

WhiteHyun committed:
:truck: 파일명 변경

WhiteHyun committed:
:cycle: ClockLayer 리팩토링

- viewDidLoad에서 layoutIfNeeded 메소드를 적용한 이후 frame 설정하도록 변경
    - frame은 한 번만 설정해도 되기 때문에 이전처럼 계속 재할당할 필요가 없기 때문

WhiteHyun committed:
:bug: 시간 레이블 버그 수정

- 백그라운드 이후 포어그라운드가 되었을 때 시간이 바뀌지 않던 버그 수정

WhiteHyun committed:
:cycle: view 계층구조에 맞추어 서브뷰를 수정
